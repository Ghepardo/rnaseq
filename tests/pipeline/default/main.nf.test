nextflow_pipeline {

    name "default"
    script "main.nf"

    test("default test config") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 197

            assert snapshot(
                path("$outputDir/salmon/salmon.merged.gene_counts_length_scaled.rds"),
                path("$outputDir/salmon/salmon.merged.gene_counts_length_scaled.tsv"),
                path("$outputDir/salmon/salmon.merged.gene_counts.rds"),
                path("$outputDir/salmon/salmon.merged.gene_counts_scaled.rds"),
                path("$outputDir/salmon/salmon.merged.gene_counts_scaled.tsv"),
                path("$outputDir/salmon/salmon.merged.gene_counts.tsv"),
                path("$outputDir/salmon/salmon.merged.gene_tpm.tsv"),
                path("$outputDir/salmon/salmon.merged.transcript_counts.rds"),
                path("$outputDir/salmon/salmon.merged.transcript_counts.tsv"),
                path("$outputDir/salmon/salmon.merged.transcript_tpm.tsv")
            ).match()

            assert new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_1_fastqc.html").exists()
            assert new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_1_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_2_fastqc.html").exists()
            assert new File("$outputDir/fastqc/RAP1_IAA_30M_REP1_2_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/RAP1_UNINDUCED_REP1_fastqc.html").exists()
            assert new File("$outputDir/fastqc/RAP1_UNINDUCED_REP1_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/RAP1_UNINDUCED_REP2_fastqc.html").exists()
            assert new File("$outputDir/fastqc/RAP1_UNINDUCED_REP2_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/WT_REP1_1_fastqc.html").exists()
            assert new File("$outputDir/fastqc/WT_REP1_1_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/WT_REP1_2_fastqc.html").exists()
            assert new File("$outputDir/fastqc/WT_REP1_2_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/WT_REP2_1_fastqc.html").exists()
            assert new File("$outputDir/fastqc/WT_REP2_1_fastqc.zip").exists()
            assert new File("$outputDir/fastqc/WT_REP2_2_fastqc.html").exists()
            assert new File("$outputDir/fastqc/WT_REP2_2_fastqc.zip").exists()

            assert new File("$outputDir/multiqc/star_salmon/multiqc_report.html").exists()

            assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.markdup.sorted.bam").exists()
            assert new File("$outputDir/star_salmon/RAP1_IAA_30M_REP1.markdup.sorted.bam.bai").exists()
            assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.markdup.sorted.bam").exists()
            assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP1.markdup.sorted.bam.bai").exists()
            assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.markdup.sorted.bam").exists()
            assert new File("$outputDir/star_salmon/RAP1_UNINDUCED_REP2.markdup.sorted.bam.bai").exists()
            assert new File("$outputDir/star_salmon/WT_REP1.markdup.sorted.bam").exists()
            assert new File("$outputDir/star_salmon/WT_REP1.markdup.sorted.bam.bai").exists()
            assert new File("$outputDir/star_salmon/WT_REP2.markdup.sorted.bam").exists()
            assert new File("$outputDir/star_salmon/WT_REP2.markdup.sorted.bam.bai").exists()

            assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_1.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_2.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP1.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP2.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/WT_REP1_1.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/WT_REP1_2.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/WT_REP2_1.fastq.gz_trimming_report.txt").exists()
            assert new File("$outputDir/trimgalore/WT_REP2_2.fastq.gz_trimming_report.txt").exists()
        }
    }

}
