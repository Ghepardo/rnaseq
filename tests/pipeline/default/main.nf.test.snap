{
    "default test config": {
        "content": [
            "salmon.merged.gene_counts_length_scaled.rds:md5,f8003e4bb35be88be899b101278b649a",
            "salmon.merged.gene_counts_length_scaled.tsv:md5,156aa4f17a9bca0c02f6ced658d1441d",
            "salmon.merged.gene_counts.rds:md5,5cd138a017b614b87332ab49cbe52cbf",
            "salmon.merged.gene_counts_scaled.rds:md5,db353b0aa735f82fa191fc60488bea57",
            "salmon.merged.gene_counts_scaled.tsv:md5,22da1c1e76dd4e015e19b450feeb3843",
            "salmon.merged.gene_counts.tsv:md5,d8a058897b775f3f09eb71ee4b62f606",
            "salmon.merged.gene_tpm.tsv:md5,db3651b0cf0583493c578f9b24c2f1c8",
            "salmon.merged.transcript_counts.rds:md5,45574d7d5a63a4f690d532b0247dc29a",
            "salmon.merged.transcript_counts.tsv:md5,05fe3760f6dbdde83e358a443f91528b",
            "salmon.merged.transcript_tpm.tsv:md5,8b071524cbdff06a2d1cf0900608af18"
        ],
        "timestamp": "2022-11-07T15:06:34+0000"
    }
}